Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,25 +144,67 @@ void HiresTexture::Update()
       }
     }
 
-    s_textureCacheAbortLoading.Clear();
+    s_textureCacheAbortPrefetching.Clear();
     s_prefetcher = std::thread(Prefetch);
   }
 }
 
 void HiresTexture::Prefetch()
 {
+  // prefetch() is running in a tight loop, and it's using that to allow "at least" 1 Search()
+  // run per cycle
```

Comment:
What this is supposed to mean isn't clear at all. It would help if you put the comment next to the relevant code instead of at the top of the whole function, and it would also help if you mentioned that this concerns Search calls that are made from another thread. And why is "at least" in quotation marks?

Reply:
Again, that is a chat log from JonnyH afaik figuring how it's suppose to work, his sentence made more sense to me than other comments in the code, afaik this is obsolete with SOIL removal, but I still see some locks staying ...

Reply:
If I'm understanding it correctly, the comment is true even after SOIL is removed. But since this comment is so hard to understand due to a complete lack of the context that was available when JonnyH said this on IRC, this comment either needs to be rewritten or removed.

Reply:
That's why I put it in I thought it would be useful, and can later decide what to do with it, he was figuring it out for 20 minutes until he concluded with this. Since I'm not experienced with this deeply, could you try to figure out a compromise by finding a suitable location and rewrite for this to save someone in the future potential 20-30 minutes of figuring out.